src/pull.go

```diff
@@ -82,6 +92,76 @@ func (g *Commands) Pull(byId bool) error {
 	return g.playPullChanges(nonConflicts, g.opts.Exports, opMap)
 }
 
+func autoRenameClashes(g *Commands, clashes []*Change) error {
+	type renameOp struct {
+		newName      string
+		change       *Change
+		originalPath string
+	}
+	clashesMap := map[string][]*Change{}
+
+	for _, clash := range clashes {
+		group := clashesMap[clash.Path]
+		if clash.Src == nil {
+			continue
+		}
+		clashesMap[clash.Path] = append(group, clash)
+	}
+
+	renames := make([]renameOp, 0, len(clashesMap)) // we will have at least len(clashesMap) renames
```

Comment:
I can't see how this could introduce bugs. The slice has still len==0 when it's created it's that the capacity is not 0 so first len(clashesMap) appends don't have to do memory reallocation. But I'm new to go so maybe I'm missing something here?